### DISCUSSION TITLE
How can I have the record data in the static form?

### DISCUSSION THREAD

**Discussion by NicolasSacC** [filamentphp/filament#4137] [Help] [answered]
Hi!

I have been trying to get the current record when I am in the "edit" page. 

Basically, I have "PickupRequests". In my form, I want to assign requests to the model I am currently creating or editing. These requests must not be already owned by another model.

When creating, it goes well because I can just take every unassigned PickupRequest: 
```
                Forms\Components\MultiSelect::make('requests')
                    ->label('Demandes de ramassage disponibles')
                    ->options(PickupRequest::active()->unassigned()->pluck('uniqid', 'id'))
                    ->visibleOn('create'),
```

However, when editing, I have no idea how I can get the current record, either in that function or in another, to create the form properly. I need to get all unassigned or current models assigned to that specific model:
```
                Forms\Components\MultiSelect::make('requests')
                    ->label('Demandes de ramassage disponibles')
                    ->options(PickupRequest::active()->->where(function ($query) {
                        $query->unassigned()->orWhere('route_id', /* HOW CAN I GET CURRENT RECORD ID HERE? */)
                    })->pluck('uniqid', 'id'))
                    ->visibleOn('edit'),
```
How can I get the record in the "options" for the query?

Also, if anyone has an idea on how I can manage a "HasMany" using only MultiSelect (since I don't want to create, just assign), I would be happy to hear your ideas!

**Comment by danharrin** [comment, marked as answer]
`->options(fn (Model $record) => ...)`

https://filamentphp.com/docs/2.x/forms/advanced#using-closure-customisation
